### DISCUSSION TITLE
Logging doesn't work

### DISCUSSION THREAD

**Discussion by bokolob** [miguelgrinberg/python-socketio#732] [Q&A]
I've got the next error when restart redis server

Apr 10 22:07:48 dj tracking_server.py[18853]: ERROR:tracking_server:Emit failed 'bool' object has no attribute 'error'
Apr 10 22:07:48 dj tracking_server.py[18853]: Traceback (most recent call last):
Apr 10 22:07:48 dj tracking_server.py[18853]:   File "/opt/venvs/tracker-backend/lib/python3.6/site-packages/socketio/asyncio_redis_manager.py", line 76, in _publish
Apr 10 22:07:48 dj tracking_server.py[18853]:     pickle.dumps(data))
Apr 10 22:07:48 dj tracking_server.py[18853]:   File "/opt/venvs/tracker-backend/lib/python3.6/site-packages/aioredis/commands/pubsub.py", line 14, in publish
Apr 10 22:07:48 dj tracking_server.py[18853]:     return self.execute(b'PUBLISH', channel, message)
Apr 10 22:07:48 dj tracking_server.py[18853]:   File "/opt/venvs/tracker-backend/lib/python3.6/site-packages/aioredis/commands/__init__.py", line 51, in execute
Apr 10 22:07:48 dj tracking_server.py[18853]:     return self._pool_or_conn.execute(command, *args, **kwargs)
Apr 10 22:07:48 dj tracking_server.py[18853]:   File "/opt/venvs/tracker-backend/lib/python3.6/site-packages/aioredis/connection.py", line 322, in execute
Apr 10 22:07:48 dj tracking_server.py[18853]:     raise ConnectionClosedError(msg)
Apr 10 22:07:48 dj tracking_server.py[18853]: aioredis.errors.ConnectionClosedError: Reader at end of file
Apr 10 22:07:48 dj tracking_server.py[18853]: During handling of the above exception, another exception occurred:
Apr 10 22:07:48 dj tracking_server.py[18853]: Traceback (most recent call last):
Apr 10 22:07:48 dj tracking_server.py[18853]:   File "/opt/venvs/tracker-backend/bin/tracking_server.py", line 76, in notify_about_new_coordinates
Apr 10 22:07:48 dj tracking_server.py[18853]:     room="__device_" + str(item))
Apr 10 22:07:48 dj tracking_server.py[18853]:   File "/opt/venvs/tracker-backend/lib/python3.6/site-packages/socketio/asyncio_pubsub_manager.py", line 69, in emit
Apr 10 22:07:48 dj tracking_server.py[18853]:     'host_id': self.host_id})
Apr 10 22:07:48 dj tracking_server.py[18853]:   File "/opt/venvs/tracker-backend/lib/python3.6/site-packages/socketio/asyncio_redis_manager.py", line 79, in _publish
Apr 10 22:07:48 dj tracking_server.py[18853]:     self._get_logger().error('Cannot publish to redis... '
Apr 10 22:07:48 dj tracking_server.py[18853]: AttributeError: 'bool' object has no attribute 'error'

**Comment by miguelgrinberg** [comment]
My guess is that you are creating your manager object and passing `logger=True`. You are confusing the documentation for the logger argument in the Server object which accepts a boolean value, the manager does not, you have to pass a logger instance or else remove this argument.

**Comment by bokolob** [comment]
Probably yes, I came up with this idea, will check it later.

Thanks!
